Add Project Card Support

jcudit opened the pull request:
Fixes https://github.com/terraform-providers/terraform-provider-github/issues/433

anGie44 commented on github/resource_github_project_card.go:
```diff
@@ -0,0 +1,140 @@
+		Update: resourceGithubProjectCardUpdate,
+		Delete: resourceGithubProjectCardDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
```
we'll need a custom import fn to set the `card_id`

anGie44 commented on github/resource_github_project_card_test.go:
```diff
@@ -0,0 +1,129 @@
+						Note: updatedNote,
+					}),
+				),
+			},
```
we'll need an ImportStep here to verify but I believe the resource's Importer func will need an update as well

jcudit commented:
The following acceptance test failure was introduced:
```
--- FAIL: TestAccGithubProjectCard_basic (13.38s)
    testing.go:654: Step 2 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) {
        }
        
        
        (map[string]string) (len=1) {
         (string) (len=9) "column_id": (string) (len=7) "9220014"
        }
        
FAIL
FAIL	github.com/terraform-providers/terraform-provider-github/github	13.397s
FAIL
```

Not sure if this is a good candidate for `ImportStateVerifyIgnore` or if I have done something wrong with the import configuration.

jcudit commented:
@anGie44 this is ready for review again after addressing your feedback.  I ended up using `ImportStateVerifyIgnore` to work around [this error](https://github.com/terraform-providers/terraform-provider-github/pull/460#issuecomment-629659922).  Let me know if you have any other ideas.

anGie44 commented on github/resource_github_project_card.go:
```diff
@@ -0,0 +1,161 @@
+	}
+
+	d.Set("note", card.GetNote())
+	d.Set("column_id", columnID)
```
looks like this assignment is where the import step is failing b/c here `columnID` is an int64 while the schema is expecting a String so the field doesn't get populated.if you pass the string from L96, it should work as expected

anGie44 commented on github/resource_github_project_card.go:
```diff
@@ -0,0 +1,161 @@
+	return nil
+}
+
+func resourceGithubProjectCardImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```
this works as expected but i'm wondering if by adding import support to this resource, we could make things a little easier by setting the resource's ID to the card_id (thereby removing the `card_id` field from the schema) and set the `node_id` as it's own field in the schema? that way the Importer func would just be a `ImportStatePassThrough` and the `id` can be an exported attribute

anGie44 replied:
ah ignore above comment -- definitely seems useful to store the node_id to help transition between v3 and v4 of the github API

anGie44 commented on github/resource_github_project_card.go:
```diff
@@ -0,0 +1,161 @@
+	d.SetId(card.GetNodeID())
+	d.Set("card_id", cardID)
+
+	return []*schema.ResourceData{d}, resourceGithubProjectCardRead(d, meta)
```
i think usually in these import funcs, the final return just needs the resource data and an explicit `nil` error as the `Read` method will implicitly be called

jcudit commented:
Rebased this and added passing tests.
